// image-overlay-react/dist/index.modern.js
import React from 'react';

var styles = {"container":"_1Lxpd","image":"_2hdkJ","overlay":"_3qw0K","left":"_2lGdA","right":"_2ehri","top":"_vpxAd","bottom":"_2BlAx","text":"_5bXm4"};

var ImgOverlay = function ImgOverlay(_ref) {
  var children = _ref.children,
      imgSrc = _ref.imgSrc,
      bgColor = _ref.bgColor,
      position = _ref.position,
      width = _ref.width,
      height = _ref.height,
      fSize = _ref.fSize,
      fColor = _ref.fColor;
  return /*#__PURE__*/React.createElement("div", {
    className: styles.container,
    style: {
      width: width,
      height: height
    }
  }, /*#__PURE__*/React.createElement("img", {
    src: imgSrc,
    alt: "Avatar",
    className: styles.image
  }), /*#__PURE__*/React.createElement("div", {
    className: styles.overlay + " " + styles[position],
    style: {
      backgroundColor: bgColor
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.text,
    style: {
      fontSize: fSize,
      color: fColor
    }
  }, children)));
};

export { ImgOverlay };
